### PR TITLE
Add basic HostedCluster cel immutability

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Add basic HostedCluster cel immutability

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Ref https://issues.redhat.com/browse/HOSTEDCP-582

Supersedes
https://github.com/openshift/hypershift/pull/1900
https://github.com/openshift/hypershift/pull/1911

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by csrwng — COMMENTED** [review]
a couple of comments

File: api/v1beta1/hostedcluster_types.go [hunk 1]
```diff
@@ -132,6 +132,8 @@ const (
 )
 
 // HostedClusterSpec is the desired behavior of a HostedCluster.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.clusterID) || has(self.clusterID)", message="ClusterID is required once set"
```
[the remark this answers is not in the document]
Comment (enxebre) [reply api/v1beta1/hostedcluster_types.go:135]: Took it from https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification

Basically saying "it's ok to not be set"

File: api/v1beta1/hostedcluster_types.go [hunk 2]
```diff
@@ -214,6 +222,8 @@ type HostedClusterSpec struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default={managementType: "Managed", managed: {storage: {type: "PersistentVolume", persistentVolume: {size: "4Gi"}}}}
 	// +immutable
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="immutable is immutable"
```
Comment (relyt0925) [inline api/v1beta1/hostedcluster_types.go:225]: Not sure this should be immutable.....

Comment (relyt0925) [reply api/v1beta1/hostedcluster_types.go:225]: secret names can potentially change in the unmanaged chase

**Comment by relyt0925** [comment]
Before we move forward with this: we need to get with the IBM team and see what fields are immutable in the general case and not and see if this is the case. Right now I am not sure on impacts and believe we need @hasueki and team to evaluate (which should be behind in priority in initially getting 4.12 operational for an IBM perspective from the teams point of view)

**Pull request closed without merging**